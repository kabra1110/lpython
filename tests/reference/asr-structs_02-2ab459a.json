{
    "basename": "asr-structs_02-2ab459a",
    "cmd": "lpython --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/structs_02.py",
    "infile_hash": "281bc75fb743f18fb6f011b278d7ab8103cc688f5856a59cb1f54895",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-structs_02-2ab459a.stdout",
    "stdout_hash": "1d4b53a95e74aa0e77fd05e9a1935431975ac4c53bedfc0812d630c5",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}