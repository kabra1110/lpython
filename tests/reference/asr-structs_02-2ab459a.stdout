(TranslationUnit
    (SymbolTable
        1
        {
            _global_symbols:
                (Module
                    (SymbolTable
                        8
                        {
                            A:
                                (StructType
                                    (SymbolTable
                                        2
                                        {
                                            x:
                                                (Variable
                                                    2
                                                    x
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            y:
                                                (Variable
                                                    2
                                                    y
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Real 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    A
                                    []
                                    [x
                                    y]
                                    Source
                                    Public
                                    .false.
                                    .false.
                                    [(())
                                    (())]
                                    ()
                                    ()
                                ),
                            _lpython_main_program:
                                (Function
                                    (SymbolTable
                                        7
                                        {
                                            
                                        })
                                    _lpython_main_program
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        []
                                        .false.
                                    )
                                    [g]
                                    []
                                    [(SubroutineCall
                                        8 g
                                        ()
                                        []
                                        ()
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            f:
                                (Function
                                    (SymbolTable
                                        3
                                        {
                                            a:
                                                (Variable
                                                    3
                                                    a
                                                    []
                                                    In
                                                    ()
                                                    ()
                                                    Default
                                                    (CPtr)
                                                    ()
                                                    BindC
                                                    Public
                                                    Required
                                                    .true.
                                                ),
                                            a1:
                                                (Variable
                                                    3
                                                    a1
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Struct
                                                        8 A
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            a2:
                                                (Variable
                                                    3
                                                    a2
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Pointer
                                                        (Struct
                                                            8 A
                                                        )
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            x:
                                                (Variable
                                                    3
                                                    x
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            y:
                                                (Variable
                                                    3
                                                    y
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Real 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    f
                                    (FunctionType
                                        [(CPtr)]
                                        ()
                                        BindC
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        []
                                        .false.
                                    )
                                    []
                                    [(Var 3 a)]
                                    [(=
                                        (Var 3 a1)
                                        (StructTypeConstructor
                                            8 A
                                            [((IntegerConstant 3 (Integer 4)))
                                            ((Cast
                                                (RealConstant
                                                    3.250000
                                                    (Real 8)
                                                )
                                                RealToReal
                                                (Real 4)
                                                (RealConstant
                                                    3.250000
                                                    (Real 4)
                                                )
                                            ))]
                                            (Struct
                                                8 A
                                            )
                                            ()
                                        )
                                        ()
                                    )
                                    (=
                                        (Var 3 a2)
                                        (GetPointer
                                            (Var 3 a1)
                                            (Pointer
                                                (Struct
                                                    8 A
                                                )
                                            )
                                            ()
                                        )
                                        ()
                                    )
                                    (Print
                                        ()
                                        [(Var 3 a2)
                                        (GetPointer
                                            (Var 3 a1)
                                            (Pointer
                                                (Struct
                                                    8 A
                                                )
                                            )
                                            ()
                                        )]
                                        ()
                                        ()
                                    )
                                    (=
                                        (Var 3 x)
                                        (StructInstanceMember
                                            (Var 3 a2)
                                            2 x
                                            (Integer 4)
                                            ()
                                        )
                                        ()
                                    )
                                    (=
                                        (Var 3 y)
                                        (StructInstanceMember
                                            (Var 3 a2)
                                            2 y
                                            (Real 4)
                                            ()
                                        )
                                        ()
                                    )
                                    (Assert
                                        (IntegerCompare
                                            (Var 3 x)
                                            Eq
                                            (IntegerConstant 3 (Integer 4))
                                            (Logical 4)
                                            ()
                                        )
                                        ()
                                    )
                                    (Assert
                                        (RealCompare
                                            (Cast
                                                (Var 3 y)
                                                RealToReal
                                                (Real 8)
                                                ()
                                            )
                                            Eq
                                            (RealConstant
                                                3.250000
                                                (Real 8)
                                            )
                                            (Logical 4)
                                            ()
                                        )
                                        ()
                                    )
                                    (CPtrToPointer
                                        (Var 3 a)
                                        (Var 3 a2)
                                        ()
                                        ()
                                    )
                                    (Print
                                        ()
                                        [(Var 3 a)
                                        (Var 3 a2)
                                        (GetPointer
                                            (Var 3 a1)
                                            (Pointer
                                                (Struct
                                                    8 A
                                                )
                                            )
                                            ()
                                        )]
                                        ()
                                        ()
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            g:
                                (Function
                                    (SymbolTable
                                        4
                                        {
                                            b:
                                                (Variable
                                                    4
                                                    b
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (CPtr)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    g
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        []
                                        .false.
                                    )
                                    [f]
                                    []
                                    [(=
                                        (Var 4 b)
                                        (PointerNullConstant
                                            (CPtr)
                                        )
                                        ()
                                    )
                                    (SubroutineCall
                                        8 f
                                        ()
                                        [((Var 4 b))]
                                        ()
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                )
                        })
                    _global_symbols
                    []
                    .false.
                    .false.
                ),
            main_program:
                (Program
                    (SymbolTable
                        6
                        {
                            _lpython_main_program:
                                (ExternalSymbol
                                    6
                                    _lpython_main_program
                                    8 _lpython_main_program
                                    _global_symbols
                                    []
                                    _lpython_main_program
                                    Public
                                )
                        })
                    main_program
                    [_global_symbols]
                    [(SubroutineCall
                        6 _lpython_main_program
                        ()
                        []
                        ()
                    )]
                )
        })
    []
)
